### DISCUSSION TITLE
linkplay devices

### DISCUSSION THREAD

**Discussion by seancmalloy** [music-assistant/support#1432] [Q&A] [answered]
i have a up2stream mini on my network using HAC linkplay in configurations.yaml. it works with the integrated media of ha but ma beta doesnt see it. actually it only sees dlna device. anyone know why? i dont see any errors.

media_player:
    - platform: linkplay
      host: 172.24.1.21
      name: Up2Stream Mini
      uuid: 'FF31F09E028AD0094D49E143'
      sources: {}
      common_sources: !include linkplay.yaml

![Screenshot (9)](https://github.com/music-assistant/hass-music-assistant/assets/23019226/a7bb3036-9b08-4d41-bafa-156e0e39fb89)

![Screenshot (10)](https://github.com/music-assistant/hass-music-assistant/assets/23019226/c1fa82ef-6d7e-4dca-a988-05ed1fe23e1b)

**Comment by OzGav** [comment]
Native link play devices are not yet supported. It is on the roadmap. ETA unknown

upvote here https://github.com/orgs/music-assistant/discussions/1200#discussioncomment-5658820

**Comment by seancmalloy** [comment]
the up2stream mini has airplay support. has anyone out there been able to add one as airplay device? ive tried airsonos add-on but receive http pico errors

**Reply by marcelveldt** [reply, marked as answer]
The up2stream (or any other linkplay based device) should work fine with both DLNA and Airplay within MA. The only part about the feature on the roadmap is to be able to control more advanced features of the linkplay chip, such as sync.

If you player is not detected in MA by default, make sure you read all our recommendations about network setup, such as making sure that MA/HA is on the same subnet as those devices.

**Reply by seancmalloy** [reply]
Ok. will do. thanx

**Comment by seancmalloy** [comment]
update - so i removed linkplay HAC and config. i reset my up2stream and reconnected it, and then connected all 3 of my airplay devices. but i when restart ha they disappear: until i re add up2stream again.  strange behavior 

i know that  you are supposed to be on the same subnet but i am able to play audio on airplay device that is on 172.24.x.x (hostapd on ha) on ha which is eth0 192.168.x.x and ma which is 172.31.x.x. audio stream doesnt work with up2stream but does with my onlink speaker and homepod speaker. all i did is add static routes in 192.68.. router to 172.24.. and 172.31.. via 192 address of ha. and ha seem to have all routes. not sure why up2stream doesnt work when others do. ???

*** Im not asking for network troubleshooting help, just stating my situation *** 

![Screenshot (12)](https://github.com/music-assistant/support/assets/23019226/968dcd9d-9934-45be-9e93-3317c2c77c0b)

**Comment by soulstyle** [comment]
@seancmalloy Did you ever get the Up2Stream device to work properly? I'm also running into problems, disconncetions, etc. Right now they are not showing up at all as Link Play devices...

**Reply by seancmalloy** [reply]
I ditched up2stream & link play - moved to ESP32-Audio-Kit dev board & Squeezelite-esp32 - this is an old project.

You can use up2stream as AirPlay - should sync multiroom
